model/fv_sg.F90

```diff
@@ -1721,6 +1721,10 @@ subroutine neg_adj3(is, ie, js, je, ng, kbot, hydrostatic, peln, delz, pt, dp,
              qv2(i,j) = qv2(i,j) - dq
              pt2(i,j) = pt2(i,j) + dq*(icpk(i,j)+lcpk(i,j))
         endif
+!tgs--  Give up and set graupel to zero 
```

Comment:
I would be extremely careful when doing this since this does not conserve water. Perhaps a warning message (with location) should be printed out whenever the adjustment is done, at least if qg2(i,j) < -1.e-20? This would avoid triggering the filling due to -eps values due to machine rounding.

Also, in what circumstances do these negatives appear?

Reply:
@lharris4 Lucas, I had a communication with you on this a couple months ago. The negative hydrometeors appear after vertical remapping. Call to neg_adj3 fixes most of it, but with Thompson MP that I was testing, negative graupel was still present after neg_adj3. I discovered that there is either no other hydrometeors around these points or the values of other positive hydrometeors around it are not sufficient to cover the negative graupel. Even with positive definite advection option for tracers, this was still happening. The change in the [model/fv_dynamics.F90] adds the diagnostics to find the location of negative hydrometeors.